### ISSUE TITLE
Argument that is both boolean and string (git exec-path option)

### ISSUE THREAD

**Issue by voieducode** [docopt/docopt.net#114]
Found an odd scenario in the git command line. Not sure how to handle this.

```
Git exec path oddity.

Usage:
 git [--exec-path[=<path>]] <command> [<args>]

Options:
  --exec-path[=<path>]
       Path to wherever your core Git programs are installed. This
       can also be controlled by setting the GIT_EXEC_PATH
       environment variable. If no path is given, git will print the
       current setting and then exit.

```

**Comment by atifaziz** [comment]
I don't think we can expect that to work since the usage isn't valid [per the docopt spec to date](http://docopt.org/); options cannot have optional values themselves. There's even [an issue open](https://github.com/docopt/docopt/issues/100) that specifically calls out the above case. What's unfortunate is that there's no error for a non-conforming usage so one is mislead to think that it will work, but in the above case, `=<path>` is seen as a _command_ ([see try.docopt.org](http://try.docopt.org/?doc=Git+exec+path+oddity.%0D%0A%0D%0AUsage%3A%0D%0A+git+%5B--exec-path%5B%3D%3Cpath%3E%5D%5D+%3Ccommand%3E+%5B%3Cargs%3E%5D%0D%0A%0D%0AOptions%3A%0D%0A++--exec-path%5B%3D%3Cpath%3E%5D%0D%0A+++++++Path+to+wherever+your+core+Git+programs+are+installed.+This%0D%0A+++++++can+also+be+controlled+by+setting+the+GIT_EXEC_PATH%0D%0A+++++++environment+variable.+If+no+path+is+given%2C+git+will+print+the%0D%0A+++++++current+setting+and+then+exit.%0D%0A%0D%0A&argv=command+arg)). It's the same for docopt.net, which builds the following pattern tree:

    Required:
    Required:
        Optional:
        Option(,--exec-path,0,False)
        Optional:
            Command(=<path>, False)
        Argument(<command>, )
        Optional:
        Argument(<args>, )
